cascade/dynamics.py

```diff
@@ -0,0 +1,136 @@
+"""Interface to run a dynamics protocol using a learned forcefield"""
+from pathlib import Path
+from typing import Any, Callable, Optional
+from dataclasses import dataclass, field
+from tempfile import TemporaryDirectory
+from uuid import uuid4
+
+from ase import Atoms
+from ase.io import Trajectory, read
+from ase.optimize.optimize import Dynamics, Optimizer
+
+from cascade.learning.base import BaseLearnableForcefield, State
+
+
+# TODO (wardlt): Consider having the state of the `Dynamics` class stored here. Some dynamics classes (e.g., NPT) have state
+@dataclass
+class Progress:
+    """The progress of an atomic state through a dynamics protocol"""
+
+    atoms: Atoms
+    """Current atomic structure"""
+    name: str = field(default_factory=lambda: str(uuid4()))
+    """Name assigned this trajectory. Defaults to a UUID"""
+    stage: int = 0
+    """Current stage within the overall :class:`DynamicsProtocol`."""
+    timestep: int = 0
```

Comment:
should there also be a property to track how far the auditor has traversed this trajectory?

Reply:
(could be added later when we implement the prototype)

Reply:
I'm not sure, so maybe we address in the next PR